### PR TITLE
[CIS-1008] Fix appearance of only emoji messages

### PR DESCRIPTION
**Pull request by SagarSDagdu**
### What does this PR do
- This PR fixes the appearance of messages which are only emojis. Such messages are shown in a larger font for up to 3 emojis.

Before             |  After
:-------------------------:|:-------------------------:
  |

### CODE REVIEW

File: Sources/StreamChatUI/Utils/String+Extensions.swift [hunk 1]
```diff
@@ -8,3 +8,42 @@ extension StringProtocol {
...
+             0x2600...0x26ff, // Misc symbols
+             0x2700...0x27bf, // Dingbats
+             0xe0020...0xe007f, // Tags
+             0xfe00...0xfe0f, // Variation Selectors
+             0x1f900...0x1f9ff, // Supplemental Symbols and Pictographs
+             127_000...127_600, // Various asian characters
+             65024...65039, // Variation selector
+             9100...9300, // Misc items
+             8400...8447: // Combining Diacritical Marks for Symbols
+            return true
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/Utils/String+Extensions.swift:44]: Where's does this come from? :)

Comment (SagarSDagdu) [reply Sources/StreamChatUI/Utils/String+Extensions.swift]: Updated the source code to include some enhancements that were introduced in Swift 5. Also updated the relevant references

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift [hunk 2]
```diff
@@ -36,8 +36,11 @@ open class _ChatMessageLayoutOptionsResolver<ExtraData: ExtraDataTypes> {
             messageIndexPath: indexPath,
             messages: messages
         )
+        
+        let isMessageOnlyOneEmoji = message.textContent?.isSingleEmoji == true && message.attachmentCounts.isEmpty == true
```
[the remark this answers is not in the document]
Comment (SagarSDagdu) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift]: Thanks for this feedback! Updated the code, currently, up to 3 emojis will be rendered in large font

File: Sources/StreamChatUI/Utils/String+Extensions_Tests.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+@testable import StreamChatUI
+import XCTest
+
+class String_Extensions_Tests: XCTestCase {
```
Comment (evsaev) [inline Sources/StreamChatUI/Utils/String+Extensions_Tests.swift:9]: nit: a true positive case for `containsOnlyEmoji` is not covered

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift [hunk 4]
```diff
@@ -512,6 +509,23 @@ open class _ChatMessageContentView<ExtraData: ExtraDataTypes>: _View, ThemeProvi
...
+        
+        if content?.isDeleted == true {
+            textColor = appearance.colorPalette.textDisabled
+        } else if content?.shouldRenderAsJumbomoji == true {
+            textFont = appearance.fonts.emoji
+        }
+        
+        textView?.textColor = textColor
+        textView?.font = textFont
+        textView?.text = content?.textContent
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:526]: nit: we don't extract setting a content/style for individual subview into a func. It absolutely makes sense to me but ATM we keep everything in `updateContent`

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift [hunk 5]
```diff
@@ -36,8 +36,9 @@ open class _ChatMessageLayoutOptionsResolver<ExtraData: ExtraDataTypes> {
             messageIndexPath: indexPath,
             messages: messages
         )
-
-        var options: ChatMessageLayoutOptions = [
+        
+        // Do not show bubble if the message is to be rendered as large emoji
+        var options: ChatMessageLayoutOptions = message.shouldRenderAsJumbomoji ? [] : [
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift:41]: nit: as soon as the bubble is no longer a default option, I'd handle it in a plain `if` as we do for other options
```swift
var options: ChatMessageLayoutOptions = [] 

if !message.shouldRenderAsJumbomoji {
   options.insert(.bubble)
}
```

**Review by evsaev — APPROVED** [review]
Great job! 👍

File: Sources/StreamChatUI/Utils/String+Extensions_Tests.swift (hunk 3 above)
Comment (SagarSDagdu) [reply Sources/StreamChatUI/Utils/String+Extensions_Tests.swift:9]: Thanks for this! Added test for `containsOnlyEmoji`

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift (hunk 5 above)
Comment (SagarSDagdu) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.swift]: Makes sense 👍🏻  Changed the code to match this style

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift (hunk 4 above)
Comment (SagarSDagdu) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift]: Got it! Moved back the text content styling to the original method.

**Review by nuno-vieira — APPROVED** [review]
LGTM!

**Comment by SagarSDagdu** [comment]
In this PR, we had to regenerate 
`Sources/StreamChatUI/ChatMessageList/ChatMessage/__Snapshots__/ChatMessageContentView_Tests/test_appearanceCustomization_usingSubclassing.small-dark.png` 
AND 
`Sources/StreamChatUI/ChatMessageList/ChatMessage/__Snapshots__/ChatMessageContentView_Tests/test_appearanceCustomization_usingUIConfig.small-dark.png`

as there was an issue in which the font of the newly generated snapshots was a bit larger than the original ones.

**Pull request merged**